cukedoctor-converter/src/main/java/com/github/cukedoctor/api/model/Feature.java

```diff
@@ -298,7 +302,25 @@ private void trySetOrder(Optional<String> order, String source) {
             try {
                 this.order = Integer.parseInt(order.get());
             } catch (Exception e) {
-                Logger.getLogger(getClass().getName()).warning(String.format("Could not get order of feature %s from %s cause: %s", name, source, e.getMessage()));
+                Logger.getLogger(getClass().getName())
+                      .warning(String.format("Could not get order of feature %s from %s cause: %s",
+                                             name,
+                                             source,
+                                             e.getMessage()));
+            }
+        }
+    }
+
+    private void trySetLanguage(Optional<String> language, String source) {
+        if (language.isPresent() && hasText(language.get())) {
+            try {
+                this.language = language.get();
+            } catch (Exception e) {
+                Logger.getLogger(getClass().getName())
+                      .warning(String.format("Could not get language of feature %s from %s cause: %s",
+                                             name,
+                                             source,
+                                             e.getMessage()));
             }
         }
```

Comment:
I know this pattern wasn't introduced by you but I'd like to avoid passing optionals as parameters. I think we could use something like this: 

```
private void setLanguageFromComment(final Comment comment) {
        comment.getLanguage()
                .filter(Assert::hasText)
                .ifPresent(this::setLanguage);
    }

    private void setLanguageFromTag(final Tag tag) {
        tag.getLanguage()
                .filter(Assert::hasText)
                .ifPresent(this::setLanguage);
    }
```

WDYT?

For setting the language I think we don't need to wrap in try catch. 

If you agree with this approach then we could do the same for setOrder:

```
private void trySetOrderFromComment(Comment comment) {
        try {
            comment.getOrder()
                    .filter(Assert::hasText)
                    .ifPresent(this::setOrder);
        } catch (Exception e) {
            Logger.getLogger(getClass().getName())
                    .warning(String.format("Could not get order of feature %s from comment cause: %s",
                            name,
                            e.getMessage()));
        }
    }

    private void trySetOrderFromTag(Tag tag) {
        try {
            tag.getOrder()
                    .filter(Assert::hasText)
                    .ifPresent(this::setOrder);
        } catch (Exception e) {
            Logger.getLogger(getClass().getName())
                    .warning(String.format("Could not get order of feature %s from tag cause: %s",
                            name,
                            e.getMessage()));
        }
    }

private void setOrder(String order) {
        this.order = Integer.parseInt(order);
    }

```

Reply:
Gool thing, that was part of my upcoming changes, I cannot agree more.
Will do.

Reply:
Since it's a technical extraction, I really just used the collections as streams:
````
  public String getLanguage() {

    if (language == null && comments != null) {
      this.language = stringExtractedFrom(comments, comment -> comment.getLanguage().orElse(null));
    }

    if (language == null && hasTags()) {
      this.language = stringExtractedFrom(getTags(), tag -> tag.getLanguage().orElse(null));
    }

    if (language == null) {
      this.language = "";
    }

    return language;
  }

  public int getOrder() {
    if (order == null && comments != null) {
      this.order = tryOrderExtractedFrom(comments, comment -> comment.getOrder().orElse(null));
    }

    if (order == null && hasTags()) {
      this.order = tryOrderExtractedFrom(getTags(), tag -> tag.getOrder().orElse(null));
    }

    if (order == null) {
      this.order = Integer.MAX_VALUE;
    }

    return order;
  }

  private <T> String stringExtractedFrom(List<T> elements, Function<T, String> extractor) {
    return elements.stream().map(extractor).filter(Assert::hasText).findFirst().orElse(null);
  }

  private <T> Integer tryOrderExtractedFrom(List<T> elements, Function<T, String> extractor) {
    String orderAsString = stringExtractedFrom(elements, extractor);
    try {
      return Integer.parseInt(orderAsString);
    } catch (Exception e) {
      log.warn(
          "Could not get order of feature {} from {} cause: {}",
          name,
          orderAsString,
          e.getMessage());
      return null;
    }
  }
````

Reply:
Yea, way better now. I liked the extraction function, nice trick.